app/src/main/cpp/src/components/transport_manager/src/bluetooth_le/ble_control_protocol.cc

```diff
@@ -0,0 +1,107 @@
+#include "transport_manager/bluetooth_le/ble_control_protocol.h"
+
+#include "utils/logger.h"
+#include "utils/jsoncpp_reader_wrapper.h"
+#include "utils/helpers.h"
+
+#include <map>
+#include <algorithm>
+
+namespace transport_manager {
+namespace transport_adapter {
+
+SDL_CREATE_LOG_VARIABLE("TransportManager")
+
+    namespace {
+        std::map<BleProtolActions, std::string> mapActionNames = {
+        {BleProtolActions::ON_DEVICE_CONNECTED, "ON_DEVICE_CONNECTED"},
+        {BleProtolActions::ON_DEVICE_DISCONNECTED, "ON_DEVICE_DISCONNECTED"}};
+
+        const char cDefaultValue[] = "_";
+        const char cActionKey[] = "action";
+        const char cNameKey[] = "name";
+        const char cAddressKey[] = "address";
+        const char cParamsKey[] = "params";
+    }
+
+    BleProtolActions BleControlProtocol::GetMessageActionType(const MsgData& message) {
+
+        utils::JsonReader reader_;
+        Json::Value value;
+        
+        if(reader_.parse((char*)(message.data()), &value)) {
+
+            auto action = value.get(cActionKey, cDefaultValue);
+
+            if(action.asString() != cDefaultValue) {
+                return GetActionFromString(action.asString());
+            }
+
+        }
+
+        return BleProtolActions::INVALID_ENUM;
+    }
+
+    BleProtolActions BleControlProtocol::GetActionFromString(const std::string& str) {
+        
+        BleProtolActions result = BleProtolActions::INVALID_ENUM;
+
+        auto findResult = std::find_if(mapActionNames.begin(), mapActionNames.end(),
+         [&](const std::pair<BleProtolActions, std::string> &pair) {
+            return pair.second == str;});
+
+        if(findResult != mapActionNames.end()) {
+            result = findResult->first;
+        }    
+
+        return result;
+    }
+
+    std::string BleControlProtocol::GetActionString(const BleProtolActions action) {
+        auto it = mapActionNames.find(action);
+        return it != mapActionNames.end() ? it->second : "UNKNOWN";
+    }
+
+    std::string BleControlProtocol::GetAddress(const MsgData& message) {
+
+        utils::JsonReader reader_;
+        Json::Value value;
+        
+        if(reader_.parse((char*)(message.data()), &value)) {
+
+            auto params = value.get(cParamsKey, Json::Value::null);
+
+            if(params != Json::Value::null) {
+
+                auto addr = params.get(cAddressKey, cDefaultValue);
+                if(addr.asString() != cDefaultValue) {
+                    return addr.asString();
+                }
+            }
+        }
+
+        return std::string();
+    }
+    
+    std::string BleControlProtocol::GetName(const MsgData& message) {
```

Comment:
@VadymLuchko actually, functions `GetName`, `GetAddress` and `GetMessageActionType` can be replaced with a single function:
```
std::string BleControlProtocol::GetValue(const std::string& key_name, const MsgData& message) {
        utils::JsonReader reader_;
        Json::Value value;

        if(reader_.parse((char*)(message.data()), &value)) {
            auto params = value.get(cParamsKey, Json::Value::null);

            if(params != Json::Value::null) {
                auto value = params.get(key_name, cDefaultValue);
                if(value.asString() != cDefaultValue) {
                    return value.asString();
                }
            }
        }

        return std::string();
    }
```
Note: for GetMessageActionType you can use `GetActionFromString` externally for returned result

Reply:
Implementations of GetName, GetAddress are very similar, so I made function GeStrFromParams to avoid code duplications.
Method GetMessageActionType - slightly different (we dont need check "params" for it), so I left it as it.

Also made GetMessageActionType and GetActionFromString public for external use;
22caff3